### PR TITLE
En_Elf

### PR DESCRIPTION
**Pull request by Kelebek1**
Before opening this PR, ensure the following:
- `./format.sh` was run to apply standard formatting.
- `make` successfully builds a matching ROM.
- No new compiler warnings were introduced during the build process.
    - Can be verified locally by running `tools/warnings_count/check_new_warnings.sh`
- New variables & functions should follow standard naming conventions.
- Comments and variables have correct spelling.
---

Documentation is from OoT.

### CODE REVIEW

File: src/overlays/actors/ovl_En_Elf/z_en_elf.c [hunk 1]
```diff
@@ -31,96 +46,1599 @@ const ActorInit En_Elf_InitVars = {
...
+        case 0:
+            thisx->room = -1;
+            EnElf_SetupAction(this, func_8088E850);
+            func_8088C51C(this, 0);
+            this->fairyFlags |= 4;
+            thisx->update = func_8089010C;
+            this->elfMsg = NULL;
+            this->unk_234 = NULL;
+            this->unk_269 = 20;
+            if ((gSaveContext.naviTimer >= 0x64C8) || (gSaveContext.naviTimer < 0xBB8)) {
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Elf/z_en_elf.c:358]:
```suggestion
            if ((gSaveContext.naviTimer >= 25800) || (gSaveContext.naviTimer < 3000)) {
```
These timers can be in dec

File: src/overlays/actors/ovl_En_Elf/z_en_elf.c [hunk 2]
```diff
@@ -31,96 +46,1599 @@ const ActorInit En_Elf_InitVars = {
...
+void func_8088D660(EnElf* this, Vec3f* targetPos, f32 arg2) {
+    f32 xVelTarget = ((targetPos->x + this->unk_224.x) - this->actor.world.pos.x) * arg2;
+    f32 zVelTarget = ((targetPos->z + this->unk_224.z) - this->actor.world.pos.z) * arg2;
+    f32 xVelDirection;
+    f32 zVelDirection;
+
+    xVelDirection = (xVelTarget >= 0.0f) ? 1.0f : -1.0f;
+    zVelDirection = (zVelTarget >= 0.0f) ? 1.0f : -1.0f;
+
+    xVelTarget = fabsf(xVelTarget);
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Elf/z_en_elf.c:516]:
```suggestion
    f32 zVelTarget = ((targetPos->z + this->unk_224.z) - this->actor.world.pos.z) * arg2;
    f32 xVelDirection = (xVelTarget >= 0.0f) ? 1.0f : -1.0f;
    f32 zVelDirection = (zVelTarget >= 0.0f) ? 1.0f : -1.0f;

    xVelTarget = fabsf(xVelTarget);
```

File: src/overlays/actors/ovl_En_Elf/z_en_elf.c [hunk 3]
```diff
@@ -31,96 +46,1599 @@ const ActorInit En_Elf_InitVars = {
...
+
+void func_8088D8D0(EnElf* this, Vec3f* arg1) {
+    f32 yVelTarget = (((Math_SinS(this->unk_246) * this->unk_250) + arg1->y) - this->actor.world.pos.y) * 0.2f;
+    f32 yVelAbs;
+    f32 yVelDirection;
+
+    yVelAbs = fabsf(yVelTarget);
+    yVelDirection = (yVelTarget >= 0.0f) ? 1.0f : -1.0f;
+
+    this->unk_246 += this->unk_24A;
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Elf/z_en_elf.c:554]:
```suggestion
    f32 yVelTarget = (((Math_SinS(this->unk_246) * this->unk_250) + arg1->y) - this->actor.world.pos.y) * 0.2f;
    f32 yVelAbs = fabsf(yVelTarget);
    f32 yVelDirection = (yVelTarget >= 0.0f) ? 1.0f : -1.0f;

    this->unk_246 += this->unk_24A;
```

File: src/overlays/actors/ovl_En_Elf/z_en_elf.c [hunk 4]
```diff
@@ -31,96 +46,1599 @@ const ActorInit En_Elf_InitVars = {
...
+    Player* refActor = GET_PLAYER(globalCtx);
+    s32 pad;
+    Player* player2 = GET_PLAYER(globalCtx);
+    f32 heightDiff;
+
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+
+    if (Rand_ZeroOne() < 0.05f) {
+        this->unk_250 = Rand_ZeroFloat(10.0f) + 10.0f;
+        this->unk_24A = (s32)Rand_ZeroFloat(1024.0f) + 0x200;
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Elf/z_en_elf.c:653]:
```suggestion
        this->unk_24A = (s32)Rand_ZeroFloat(0x400) + 0x200;
```

File: src/overlays/actors/ovl_En_Elf/z_en_elf.c [hunk 5]
```diff
@@ -31,96 +46,1599 @@ const ActorInit En_Elf_InitVars = {
...
+        if ((arrayPointerActor == NULL) || (player->unk_730 == NULL)) {
+            this->fairyFlags ^= 1;
+        }
+    } else if ((arrayPointerActor != NULL) && (player->unk_730 != NULL)) {
+        u8 temp = this->unk_269;
+        u16 targetSound = this->unk_269 == 0 ? NA_SE_PL_WALK_GROUND - SFX_FLAG : NA_SE_PL_WALK_GROUND - SFX_FLAG;
+
+        if (!temp) {
+            Audio_PlayActorSound2(&this->actor, targetSound);
+        }
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Elf/z_en_elf.c:1104]:
```suggestion
        u16 targetSfxId = this->unk_269 == 0 ? NA_SE_PL_WALK_GROUND - SFX_FLAG : NA_SE_PL_WALK_GROUND - SFX_FLAG;

        if (!temp) {
            Audio_PlayActorSound2(&this->actor, targetSfxId);
        }
```
`SfxId` is the name we use for sound effects

File: src/overlays/actors/ovl_En_Elf/z_en_elf.c [hunk 6]
```diff
@@ -31,96 +46,1599 @@ const ActorInit En_Elf_InitVars = {
...
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Elf/func_8088C858.s")
+                            case 2:
+                                func_80151938(globalCtx, 0x247);
+                                break;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Elf/func_8088C920.s")
+                            case 3:
+                                if (gSaveContext.isNight == 0) {
+                                    func_80151938(globalCtx, 0x248);
+                                } else if ((gSaveContext.time < 0x4000) && (gSaveContext.weekEventReg[74] & 0x20)) {
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Elf/z_en_elf.c:1425]:
```suggestion
                                } else if ((gSaveContext.time < CLOCK_TIME(6, 0)) && (gSaveContext.weekEventReg[74] & 0x20)) {
```

File: src/overlays/actors/ovl_En_Elf/z_en_elf.c [hunk 7]
```diff
@@ -31,96 +46,1599 @@ const ActorInit En_Elf_InitVars = {
...
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Elf/EnElf_Destroy.s")
+    if (temp_v0 != this->unk_266) {
+        this->unk_266 = temp_v0;
+        gSaveContext.naviTimer = 0;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Elf/func_8088D3EC.s")
+    if ((player->tatlTextId == 0) && (player->unk_730 == NULL)) {
+        if ((gSaveContext.naviTimer >= 0x258) && (gSaveContext.naviTimer < 0xBB9)) {
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Elf/z_en_elf.c:1466]:
```suggestion
        if ((gSaveContext.naviTimer >= 600) && (gSaveContext.naviTimer <= 3000)) {
```

File: src/overlays/actors/ovl_En_Elf/z_en_elf.c [hunk 8]
```diff
@@ -31,96 +46,1599 @@ const ActorInit En_Elf_InitVars = {
...
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Elf/func_8088D470.s")
+    if (player->tatlTextId < 0) {
+        thisx->flags |= 0x10000;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Elf/func_8088D504.s")
+    if (func_800B84D0(thisx, globalCtx)) {
+        func_8019FDC8(&D_801DB4A4, 0x6873, 0x20);
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Elf/z_en_elf.c:1476]:
```suggestion
        func_8019FDC8(&D_801DB4A4, NA_SE_VO_NA_LISTEN, 0x20);
```

File: src/overlays/actors/ovl_En_Elf/z_en_elf.c [hunk 9]
```diff
@@ -31,96 +46,1599 @@ const ActorInit En_Elf_InitVars = {
...
+    endPos.x = npcAction->unk18.x;
+    endPos.y = npcAction->unk18.y;
+    endPos.z = npcAction->unk18.z;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Elf/func_808908D0.s")
+    lerp = func_800F5A8C(npcAction->endFrame, npcAction->startFrame, globalCtx->csCtx.frames, globalCtx);
+    vec->x = ((endPos.x - startPos.x) * lerp) + startPos.x;
+    vec->y = ((endPos.y - startPos.y) * lerp) + startPos.y;
+    vec->z = ((endPos.z - startPos.z) * lerp) + startPos.z;
+}
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Elf/z_en_elf.c:1644]: Could you get this to match with the `VEC3F_LERPIMPDST` macro?

**Review by engineer124 — COMMENTED** [review]
Nice work. Just a few small comments

File: src/overlays/actors/ovl_En_Elf/z_en_elf.c [hunk 10]
```diff
@@ -37,96 +46,1592 @@ const ActorInit En_Elf_InitVars = {
...
+            s32 envAlpha;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Elf/func_8088EF18.s")
+            OPEN_DISPS(globalCtx->state.gfxCtx);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Elf/func_8088EFA4.s")
+            func_8012C94C(globalCtx->state.gfxCtx);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Elf/func_8088F214.s")
+            envAlpha = (this->timer * 50) & 0x1FF;
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_En_Elf/z_en_elf.c:1591]: Would this match?
```suggestion
            envAlpha = (this->timer * 50) % 512;
```

File: src/overlays/actors/ovl_En_Elf/z_en_elf.c [hunk 11]
```diff
@@ -37,96 +46,1592 @@ const ActorInit En_Elf_InitVars = {
...
+                    case 543:
+                    case 544:
+                    case 545:
+                    case 546:
+                    case 547:
+                    case 575:
+                    case 577:
+                    case 578:
+                    case 579:
+                    case 580:
```
Comment (hensldm) [inline src/overlays/actors/ovl_En_Elf/z_en_elf.c:1406]: These cases look like textIds so should probably be in hex.

File: src/overlays/actors/ovl_En_Elf/z_en_elf.c [hunk 12]
```diff
@@ -37,96 +46,1592 @@ const ActorInit En_Elf_InitVars = {
...
+void func_8089010C(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    EnElf* this = THIS;
+    Player* player = GET_PLAYER(globalCtx);
+    u16 temp_v0 = func_800F05C0(globalCtx);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Elf/EnElf_Destroy.s")
+    if (temp_v0 != this->unk_266) {
+        this->unk_266 = temp_v0;
+        gSaveContext.naviTimer = 0;
```
Comment (hensldm) [inline src/overlays/actors/ovl_En_Elf/z_en_elf.c:1457]: I know this wasn't you, but this should be changed to tatlTimer.

**Review by hensldm — CHANGES_REQUESTED** [review]
A few things I noticed.

File: src/overlays/actors/ovl_En_Elf/z_en_elf.c (hunk 10 above)
Comment (Kelebek1) [reply src/overlays/actors/ovl_En_Elf/z_en_elf.c:1591]: Doesn't work due to being signed.

**Review by AngheloAlf — APPROVED** [review]

**Review by hensldm — APPROVED** [review]

**Review by Kenix3 — APPROVED** [review]

**Pull request merged**